### PR TITLE
Populate signatureDocumentation

### LINKED ISSUE #85: Update indexer to populate `signature_documentation`
The scip spec introduced [`signature_documentation`](https://github.com/sourcegraph/scip/blob/main/scip.proto#L338) as a way of specifying the code signature of a symbol

As per [this](https://github.com/sourcegraph/scip/blob/main/scip.proto#L194) comment, scip-dart currently re-uses the `documentation` field for this 

Additional note, `signature_documentation` supports indexing this signature field, and allowing documentation / goto definition. Along with implementing this field, implement this functionality (which I'm not exactly sure if this is possible with the dart analyzer)

### PR DESCRIPTION
**Pull request by matthewnitschke-wk**
closes #85

### CODE REVIEW

**Comment by aviary-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: lib/src/gen/scip.pb.dart
Comment (matthewnitschke-wk) [inline lib/src/gen/scip.pb.dart:1]: generated file change

File: lib/src/gen/scip.pbjson.dart
Comment (matthewnitschke-wk) [inline lib/src/gen/scip.pbjson.dart:1]: generated file change

File: lib/src/gen/scip.proto
Comment (matthewnitschke-wk) [inline lib/src/gen/scip.proto:1]: Copy and paste update from scip's official protobuf file: https://github.com/sourcegraph/scip/blob/main/scip.proto

File: pubspec.lock [hunk 1]
```diff
@@ -7,21 +7,21 @@ packages:
...
-    version: "61.0.0"
+    version: "54.0.0"
   analyzer:
     dependency: "direct main"
     description:
       name: analyzer
       url: "https://pub.dartlang.org"
     source: hosted
-    version: "5.13.0"
+    version: "5.6.0"
```
Comment (matthewnitschke-wk) [inline pubspec.lock:17]: It looks like dependabot dart is doing some strange things

analyzer `v5.13.0` requires dart `2.19.0`, and dependabot happily bumped to that version: https://github.com/Workiva/scip-dart/pull/75

But, as can be seen from the pubspec.yaml, the min dart version is still set to `">=2.18.0 <3.0.0"`, implying dependabot is ignoring this specification

This means that when dependabot directly changes just the pubspec.lock file, its bumping the lockfile, but anytime you re-run pub get from the codebase, pub will correctly adjust the version used

File: .github/dependabot.yml [hunk 2]
```diff
@@ -1,6 +1,7 @@
 version: 2
 updates:
-  - package-ecosystem: "pub"
+  - package-ecosystem: pub
+    versioning-strategy: increase
```
Comment (matthewnitschke-wk) [inline .github/dependabot.yml:4]: A bit of an unrelated change, but dependabot's default setting is `widen` for pub

This basically means: "probably dont change the pubspec.yaml file"

We dont want this, always change the pubspec.yaml file to enforce that latest versions of pubspec files

(background issue: https://github.com/dependabot/dependabot-core/issues/4979)

**Pull request closed without merging**